src/main/java/elrond/esdt/common/TokenPayment.java

```diff
@@ -0,0 +1,75 @@
+package elrond.esdt.common;
+
+import org.apache.commons.codec.binary.Hex;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Collections;
+
+public class TokenPayment {
+    private final TokenIdentifier tokenIdentifier;
+    private final BigDecimal amountAsBigDecimal;
+    private final Integer numDecimals;
+    private final Long nonce;
+
+    public TokenPayment(TokenIdentifier identifier, BigDecimal amountAsBigDecimal, Integer numDecimals, Long nonce) {
+        this.tokenIdentifier = identifier;
+        this.amountAsBigDecimal = amountAsBigDecimal;
+        this.numDecimals = numDecimals;
+        this.nonce = nonce;
+    }
+
+    public static TokenPayment egldFromAmount(String amount) {
+        BigDecimal amountAsBigDecimal = increaseMagnitude(new BigDecimal(amount), Constants.EGLDNumDecimals);
+        return TokenPayment.egldFromBigDecimal(amountAsBigDecimal);
+    }
+
+    public static TokenPayment egldFromBigDecimal(BigDecimal amountAsBigDecimal) {
+        return new TokenPayment(Constants.EGLDIdentifier, amountAsBigDecimal, Constants.EGLDNumDecimals, 0L);
+    }
+
+    public static TokenPayment fungibleFromAmount(TokenIdentifier tokenIdentifier, String amount, Integer numDecimals) {
+        BigDecimal amountAsBigDecimal = increaseMagnitude(new BigDecimal(amount), numDecimals);
+        return TokenPayment.fungibleFromBigDecimal(tokenIdentifier, amountAsBigDecimal, numDecimals);
+    }
+
+    public static TokenPayment fungibleFromBigDecimal(TokenIdentifier tokenIdentifier, BigDecimal amountAsBigDecimal, Integer numDecimals) {
+        return new TokenPayment(tokenIdentifier, amountAsBigDecimal, numDecimals, 0L);
+    }
+
+    public static TokenPayment semiFungible(TokenIdentifier tokenIdentifier, Integer quantity, Long nonce) {
+        return new TokenPayment(tokenIdentifier, new BigDecimal(quantity.toString()), 0, nonce);
+    }
+
+    public static TokenPayment nonFungible(TokenIdentifier tokenIdentifier, Long nonce) {
```

Comment:
maybe add one static method also for metaEsdt ?

Reply:
added